### PR TITLE
Log levels

### PR DESCRIPTION
**Pull request by ianb**

### CODE REVIEW

File: addon/webextension/log.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+/* globals buildSettings */
+
+"use strict";
+
+this.log = (function () {
+  let exports = {};
+
+  const levels = ["debug", "info", "warn", "error"];
+  if (! levels.includes(buildSettings.logLevel)) {
+    console.warn("Invalid buildSettings.logLevel:", buildSettings.logLevel);
```
Comment (6a68) [inline addon/webextension/log.js]: Probably should use a template string here, was mentioned in the webext review

File: addon/webextension/log.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  }
+  let shouldLog = {};
+
+  {
+    let startLogging = false;
+    for (let level of levels) {
+      if (buildSettings.logLevel == level) {
```
Comment (6a68) [inline addon/webextension/log.js]: `===` here?

File: addon/webextension/log.js (hunk 1 above)
Comment (ianb) [inline addon/webextension/log.js]: This isn't concatenation, it's just normal console.* multiple argument usage

File: addon/webextension/log.js (hunk 2 above)
Comment (ianb) [inline addon/webextension/log.js]: Mozilla style is unclear on this, e.g., [this](https://developer.mozilla.org/en-US/docs/Mozilla/Developer_guide/Coding_Style) recommends `==`.  The code itself is kind of all over the place.

I changed, but probably best to let some future eslint rule manage this.

**Pull request merged**